src/tx.sw

```diff
@@ -0,0 +1,219 @@
+//! Transaction field getters.
+//! This will be replaced by instructions: https://github.com/FuelLabs/fuel-specs/issues/287
+library tx;
+
+use ::address::Address;
+use ::contract_id::ContractId;
+
+////////////////////////////////////////
+// Transaction fields
+////////////////////////////////////////
+
+// The transaction starts at
+// TX_START = 32 + MAX_INPUTS*(32+8) + 8 = 32 + 255 * (40) + 8 = 10240
+//
+// Note that everything when serialized is padded to word length.
+//
+// type             = TX_START +  0*WORD_SIZE = 10240 +  0*8 = 10240
+// gasPrice         = TX_START +  1*WORD_SIZE = 10240 +  1*8 = 10248
+// gasLimit         = TX_START +  2*WORD_SIZE = 10240 +  2*8 = 10256
+// bytePrice        = TX_START +  3*WORD_SIZE = 10240 +  3*8 = 10264
+// maturity         = TX_START +  4*WORD_SIZE = 10240 +  4*8 = 10272
+// scriptLength     = TX_START +  5*WORD_SIZE = 10240 +  5*8 = 10280
+// scriptDataLength = TX_START +  6*WORD_SIZE = 10240 +  6*8 = 10288
+// inputsCount      = TX_START +  7*WORD_SIZE = 10240 +  7*8 = 10296
+// outputsCount     = TX_START +  8*WORD_SIZE = 10240 +  8*8 = 10304
+// witnessesCount   = TX_START +  9*WORD_SIZE = 10240 +  9*8 = 10312
+// receiptsRoot     = TX_START + 10*WORD_SIZE = 10240 + 10*8 = 10320
+// script start     = TX_START + 11*WORD_SIZE = 10240 + 14*8 = 10352
+
+const TX_TYPE_OFFSET = 10240;
+const TX_GAS_PRICE_OFFSET = 10248;
+const TX_GAS_LIMIT_OFFSET = 10256;
+const TX_BYTE_PRICE_OFFSET = 10264;
+const TX_MATURITY_OFFSET = 10272;
+const TX_SCRIPT_LENGTH_OFFSET = 10280;
+const TX_SCRIPT_DATA_LENGTH_OFFSET = 10288;
+const TX_INPUTS_COUNT_OFFSET = 10296;
+const TX_OUTPUTS_COUNT_OFFSET = 10304;
+const TX_WITNESSES_COUNT_OFFSET = 10312;
+const TX_RECEIPTS_ROOT_OFFSET = 10320;
+const TX_SCRIPT_START_OFFSET = 10352;
+
+/// Get the transaction type.
+pub fn tx_type() -> u8 {
+    asm(r1, r2: 10240) {
+        lw r1 r2 i0;
+        r1: u8
+    }
+}
+
+/// Get the transaction gas price.
+pub fn tx_gas_price() -> u64 {
+    asm(r1, r2: 10248) {
```

Comment:
Assuming that `r2` here would eventually use `TX_GAS_PRICE_OFFSET` when possible.
Same for the other `const`s

Reply:
Yes, see commit message https://github.com/FuelLabs/sway-lib-std/pull/93/commits/618f8a65836a44fa30d7e238b2cddc34216fca34